src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts

```diff
@@ -81,9 +85,41 @@ export class PasteImageProvider implements DocumentPasteEditProvider {
 			return;
 		}
 
-		widget.attachmentModel.addContext(imageContext);
+		// Make sure to attach only new contexts
+		const currentContextIds = widget.attachmentModel.getAttachmentIDs();
+		if (currentContextIds.has(imageContext.id)) {
+			return;
+		}
+
+		const customEdit = {
+			resource: model.uri,
+			variable: imageContext,
+			undo: () => {
+				widget = this.chatWidgetService.getWidgetByInputUri(model.uri);
+				if (!widget) {
+					throw new Error('No widget found for undo');
+				}
+				widget.attachmentModel.delete(imageContext.id);
+			},
+			redo: () => {
+				widget = this.chatWidgetService.getWidgetByInputUri(model.uri);
+				if (!widget) {
+					throw new Error('No widget found for redo');
+				}
+				widget.attachmentModel.addContext(imageContext);
+			},
+			metadata: { needsConfirmation: false, label: imageContext.name }
+		};
 
-		return;
+		return {
+			edits: [{
+				insertText: '', title: 'Empty Edit', kind: new HierarchicalKind('chat.attach.image'), handledMimeType: mimeType,
```

Comment:
Make sure to fill in the `title`. This can show in the UI if there are multiple possible ways to paste